File: Data Structures & Algorithms/mergeSort/submission-1.go
// Definition for a pair.
// type Pair struct {
//     Key   int
//     Value string
// }

func mergeSort(pairs []Pair) []Pair {
    merge := func (s,m,e int) {
        L := make([]Pair,m-s+1)
        R := make([]Pair,e-m)
        copy(L,pairs[s:m+1])
        copy(R,pairs[m+1:e+1])
        i,j,k := 0,0,s
        for i < len(L) && j < len(R) {
            if L[i].Key <= R[j].Key {
                pairs[k] = L[i]
                i += 1
            } else {
                pairs[k] = R[j]
                j += 1
            }
            k += 1
        }
        for i < len(L) {
            pairs[k] = L[i]
            i += 1
            k += 1
        }
        for j < len(R) {
            pairs[k] = R[j]
            j += 1
            k += 1
        }
    }
    var dfs func(int,int)
    dfs = func (s,e int) {
        if e - s > 0 {
            m := (e+s)/2
            dfs(s,m)
            dfs(m+1,e)
            merge(s,m,e)
        }
    }
    dfs(0,len(pairs)-1)
    return pairs
}

File: Data Structures & Algorithms/quickSort/submission-1.py
# Definition for a pair.
# class Pair:
#     def __init__(self, key: int, value: str):
#         self.key = key
#         self.value = value
class Solution:
    def quickSort(self, pairs: List[Pair]) -> List[Pair]:
        return self.dfs(pairs,0,len(pairs)-1)

    def swap(self,arr,i,j):
        arr[i],arr[j] = arr[j],arr[i]

    def dfs(self,arr: List[Pair], s: int, e: int) -> List[Pair]:
        if e-s > 0:
            left = s
            for i in range(s,e):
                if arr[i].key < arr[e].key:
                    self.swap(arr,i,left)
                    left += 1
            self.swap(arr,e,left)
            self.dfs(arr,s,left-1)
            self.dfs(arr,left+1,e)
        return arr